ra/registration-authority.go

```diff
@@ -381,17 +381,41 @@ func (ra *RegistrationAuthorityImpl) UpdateAuthorization(base core.Authorization
 }
 
 // RevokeCertificate terminates trust in the certificate provided.
-func (ra *RegistrationAuthorityImpl) RevokeCertificate(cert x509.Certificate) (err error) {
+func (ra *RegistrationAuthorityImpl) RevokeCertificate(cert x509.Certificate, reasonCode int, regID *int64) (err error) {
 	serialString := core.SerialToString(cert.SerialNumber)
-	err = ra.CA.RevokeCertificate(serialString, 0)
+	err = ra.CA.RevokeCertificate(serialString, reasonCode)
```

Comment:
Point of order, why do we go through the RA if we're calling the same code in the CA?